### ISSUE TITLE
Add MP OpenAPI server.xml config to include or exclude apps (2.0+)

### ISSUE THREAD

**Issue by Azquelt** [OpenLiberty/open-liberty#28857]
## New server.xml elements:

```xml
<mpOpenAPI>​
    <include>myApp1/myModule</include>​
</mpOpenAPI>​
```

```xml
<mpOpenAPI>​
    <exclude>myInternalApp</exclude>​
</mpOpenAPI>​
```

Semantics should match the existing MP config properties for including or excluding apps: https://openliberty.io/docs/latest/documentation-openapi.html#multi-module

If either `<include>` or `<exclude>` are present in the server.xml, the MP config properties should be ignored.

**Comment by Azquelt** [comment]
We're reviewing the design of this part of the feature in #29541. We need to get the design questions resolved before we can implement this.

**Comment by Azquelt** [comment]
Tasks:
- [x] Add new metatype definitions, make sure they're marked `ibm:beta`
- [x] Have a component that reads the config - currently this is `OpenAPIEndpointManager` as we only have endpoint config, so some refactoring here is needed.
- [x] Update existing code so that config can be updated dynamically and the OpenAPI document recreated if necessary without restarting the applications.
  - Likely we need most of the change in `ApplicationRegistry` - from an initial look, I think we would want to call `ApplicationProcessor.processApplication` for each deployed application when the config changes.
  - We need to ensure that we store all the relevant info about each application when it starts so that if the config gets updated, we can redo the creation of the OpenAPI documents
- [x] Test that server.xml config is not processed if beta flag is not set (to be removed when feature is done)
- [x] Update `ModuleSelectionConfig` so that it reads server.xml config as well as MP Config and selects modules appropriately
- [x] Keep `MergeConfigTest` which tests MP Config properties, and add similar cases using the new server.xml config
- [x] Test that we ignore MP Config properties if the new server.xml config elements are used
- [x] Test that we can update the include/exclude config while the server is running and the OpenAPI document is updated

**Comment by benjamin-confino** [comment]
Note to self, also have to open a new issue for the warning messages for an included statement that doesn't match an app
